apps/web/app/api/(internal)/insights/lib/document.ts

```diff
@@ -37,7 +37,8 @@ export const createDocument = async (
     const { object } = await generateObject({
       model: llmModel,
       schema: ZGenerateDocumentObjectSchema,
-      system: `You are an XM researcher. You analyse a survey response (survey name, question headline & user answer) and generate insights from it. The insight title (1-3 words) should concisely answer the question, e.g., "What type of people do you think would most benefit" -> "Developers". You are very objective. For the insights, split the feedback into the smallest parts possible and only use the feedback itself to draw conclusions. You must output at least one insight. Always generate insights and titles in English, regardless of the input language.`,
+      system:
+        "As an XM researcher, your task is to analyze a collection of survey responses about a product or feature and distill the feedback into a concise set of high-level insights rather than creating one insight per response. Group thematically related observations into broader categories to identify key trends. For example, if users mention slow performance, loading delays, or sluggish behavior, consolidate these under a single insight titled *Performance Issues.* Similarly, feedback about the need for more integrations, multilingual support, or broader export options can be grouped under a category like *Platform Enhancements.* Any response, whether positive or negative, that directly addresses the feature in question should be included under an insight titled *Feature Feedback.* Each insight should have a clear, concise title (1–3 words) and a brief, objective description (1–2 sentences) that reflects the core theme without adding assumptions or extrapolations. For example, under the theme of *Documentation Feedback,* insights could include *Clarity Issues,* highlighting unclear or overly complex sections, *Coverage Gaps,* noting missing information like troubleshooting guides, and *Accessibility Improvements,* addressing the lack of features such as better navigation or multilingual support. This approach ensures that major trends are captured and summarized effectively while keeping the insights concise, actionable, and focused on overarching themes. Always present insights in English, regardless of the input language.",
```

Comment:
I'm not sure if the bucketing gets too generic here.
```
feedback about the need for more integrations, multilingual support, or broader export options can be grouped under a category like *Platform Enhancements.*
```

This is essentially just "Feature Request" which is also one of our categories, but if you ask the users what features are missing, get 1000 responses and all you get as an insight is "Platform Enhancements", the Formbricks won't be very happy about this summary.

We need to find a good balance between being generic enough to group appropriately and being specific enough to provide value.